pkg/devfile/parser/configurables.go

```diff
@@ -0,0 +1,218 @@
+	Ports             = "Ports"
+	Memory            = "Memory"
+	PortsDescription  = "Ports to be opened in all component containers"
+	MemoryDescription = "The Maximum memory all the component containers can consume"
+	NameDescription   = "The name of the component"
+)
+
+var (
+	supportedDevfileParameterDescriptions = map[string]string{
+		Name:   NameDescription,
+		Ports:  PortsDescription,
+		Memory: MemoryDescription,
+	}
+
+	lowerCaseDevfileParameters = util.GetLowerCaseParameters(GetDevfileSupportedParameters())
+)
+
+func (d DevfileObj) SetConfiguration(parameter string, value interface{}) error {
+
+	strValue, _ := value.(string)
+	if parameter, ok := AsDevfileSupportedParameter(parameter); ok {
+		switch parameter {
+		case "name":
+			return d.setMetadataName(strValue)
+		case "ports":
+			arrValue := strings.Split(strValue, ",")
+			return d.setPorts(arrValue...)
+		case "memory":
+			return d.setMemory(strValue)
+		}
+
+	}
+	return errors.Errorf("unknown parameter :'%s' is not a configurable parameter in the devfile", parameter)
+
+}
+
+func (d DevfileObj) DeleteConfiguration(parameter string) error {
+	if parameter, ok := AsDevfileSupportedParameter(parameter); ok {
+		switch parameter {
+		case "name":
+			return d.setMetadataName("")
+		case "ports":
+			return d.removePorts()
+		case "memory":
+			return d.setMemory("")
+		}
+	}
+	return errors.Errorf("unknown parameter :'%s' is not a configurable parameter in the devfile", parameter)
+}
+
+func (d DevfileObj) IsSet(parameter string) bool {
+
+	if parameter, ok := AsDevfileSupportedParameter(parameter); ok {
+		switch parameter {
+		case "name":
+			return d.getMetadataName() != ""
+		case "ports":
+			return d.hasPorts()
+		case "memory":
+			return d.getMemory() != ""
+		}
+	}
+	return false
+
+}
+
+func (d DevfileObj) setMetadataName(name string) error {
+	metadata := d.Data.GetMetadata()
+	d.Data.SetMetadata(name, metadata.Version)
+	return d.WriteYamlDevfile()
+}
+
+func (d DevfileObj) AddEnvVars(otherList config.EnvVarList) error {
+	components := d.Data.GetComponents()
+	for _, component := range components {
+		if component.Container != nil {
+			currentlist := config.NewEnvVarListFromDevfileEnv(component.Container.Env)
+			component.Container.Env = currentlist.Merge(otherList).ToDevfileEnv()
+			d.Data.UpdateComponent(component)
+		}
+	}
+	return d.WriteYamlDevfile()
+}
+
+func (d DevfileObj) RemoveEnvVars(keys []string) error {
+	components := d.Data.GetComponents()
+	for _, component := range components {
+		if component.Container != nil {
+
+			currentlist := config.NewEnvVarListFromDevfileEnv(component.Container.Env)
+			envList, err := config.RemoveEnvVarsFromList(currentlist, keys)
+			if err != nil {
+				return err
+			}
+			component.Container.Env = envList.ToDevfileEnv()
+			d.Data.UpdateComponent(component)
+		}
+	}
+	return d.WriteYamlDevfile()
+}
+
+func (d DevfileObj) setPorts(ports ...string) error {
+	components := d.Data.GetComponents()
```

Comment:
it is completely replacing the endpoints list, not appending
is it expected behavior?

Reply:
yes, this was thought out and finalised as the behavior.

Reply:
we might want to discuss more on that due to issue: https://github.com/openshift/odo/issues/3718
We are going to save all URL information in devfile endpoint entry, and we don't want to lose all the settings due to a call of `odo config`

Reply:
handled, now ports are appended